pkg/cmd/server/kubernetes/master/master_config.go

```diff
@@ -369,6 +375,60 @@ func BuildControllerManagerServer(masterConfig configapi.MasterConfig) (*cmapp.C
 	return cmserver, cloud, nil
 }
 
+// AddRecyclerTemplates creates a file with default recycled pod template and
+// adds it into PersistentVolumeRecyclerConfiguration if it is not set. It
+// returns name of the template file that should be deleted after controllers
+// start.
+func AddRecyclerTemplates(masterConfig configapi.MasterConfig, cmserver *cmapp.CMServer) (string, error) {
+	if cmserver.VolumeConfiguration.PersistentVolumeRecyclerConfiguration.PodTemplateFilePathHostPath != "" && cmserver.VolumeConfiguration.PersistentVolumeRecyclerConfiguration.PodTemplateFilePathNFS != "" {
+		// Both templates are set by user config, there is nothing to configure.
+		return "", nil
+	}
+
+	// At least one template is needed, fill it
+	imageTemplate := variable.NewDefaultImageTemplate()
+	imageTemplate.Format = masterConfig.ImageConfig.Format
+	imageTemplate.Latest = masterConfig.ImageConfig.Latest
+	uid := int64(0)
+
+	template := volume.NewPersistentVolumeRecyclerPodTemplate()
+	template.Namespace = "openshift-infra"
+	template.Spec.ServiceAccountName = bootstrappolicy.InfraPersistentVolumeRecyclerControllerServiceAccountName
+	template.Spec.Containers[0].Image = imageTemplate.ExpandOrDie("recycler")
+	template.Spec.Containers[0].Command = []string{"/usr/bin/openshift-recycle"}
+	template.Spec.Containers[0].Args = []string{"/scrub"}
+	template.Spec.Containers[0].SecurityContext = &kapiv1.SecurityContext{RunAsUser: &uid}
+	template.Spec.Containers[0].ImagePullPolicy = kapiv1.PullIfNotPresent
+
+	templateBytes, err := json.Marshal(template)
```

Comment:
Needs to go through our normal encoding path.  `runtime.Encode(kapi.Codecs.LegacyCodec(` would do I think.

Reply:
fixed